index.js

```diff
@@ -111,6 +116,44 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+
+        // do not look for properties on string plugins
+        if (pluginType === 'function' || pluginType === 'object') {
+          pluginAugmentsCacheKey = typeof plugin.cacheKey === 'function';
+          pluginProvidesBaseDir = typeof plugin.baseDir === 'function';
+        }
+
+        if (pluginAugmentsCacheKey) {
+          hash.plugins.push(plugin.cacheKey());
+        }
+
+        if (pluginProvidesBaseDir) {
+          var depHash = hashForDep(plugin.baseDir());
+
+          hash.plugins.push(depHash);
+        }
+
+        if (pluginType === 'string') {
+          hash.plugins.push(plugin);
+        } else if (!pluginProvidesBaseDir){
+          // prevent caching completely if the plugin doesn't provide baseDir
```

Comment:
I don't think that is legal (though I will have to check), I think (in 5.x) they need to be functions that are invoked?